services/QuillLMS/app/services/classlink_integration/user_persister.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module ClasslinkIntegration
+  class UserPersister < ApplicationService
+    attr_reader :incoming_user, :existing_provider_users, :classlink_instance_id
+
+    def initialize(incoming_user:, existing_provider_users:, classlink_instance_id:)
+      @incoming_user = incoming_user
+      @existing_provider_users = existing_provider_users
+      @classlink_instance_id = classlink_instance_id
+    end
+
+    def run
+      creatable_user_hash = UserPersister.creatable_user(incoming_user)
+
+      if existing_user
+        existing_user.user.update!(**creatable_user_hash)
+      else
+        ActiveRecord::Base.transaction do
+          User.transaction do
```

Comment:
Do we need two layers of `transaction` declarations here?

Reply:
no; removing